trace/backend_collector_test.go

```diff
@@ -264,73 +264,222 @@ func newTestCookieSpec(d *pb.Data, key string, responseCode int) *pb.Data {
+			},
+			ExpectedWitnesses: 1,
+			ExpectedLatencies: []float32{-1.0},
+		},
+		{
+			Name: "request + response fully overlapping",
+			PNTs: []akinet.ParsedNetworkTraffic{
+				request(streamID, seq, startTime, startTime),
+				response(streamID, seq, startTime, startTime),
+			},
+			ExpectedWitnesses: 1,
+			ExpectedLatencies: []float32{0.0},
+		},
+		{
+			Name: "response + request fully overlapping",
+			PNTs: []akinet.ParsedNetworkTraffic{
+				response(streamID, seq, startTime, startTime),
+				request(streamID, seq, startTime, startTime),
+			},
+			ExpectedWitnesses: 1,
+			ExpectedLatencies: []float32{0.0},
+		},
+		{
+			Name: "request + response out of order",
+			PNTs: []akinet.ParsedNetworkTraffic{
+				request(streamID, seq, startTime.Add(3*time.Millisecond), startTime.Add(4*time.Millisecond)),
+				response(streamID, seq, startTime, startTime.Add(2*time.Millisecond)),
+			},
+			ExpectedWitnesses: 1,
+			ExpectedLatencies: []float32{-4.0},
+		},
+		{
+			Name: "response + request out of order",
+			PNTs: []akinet.ParsedNetworkTraffic{
+				response(streamID, seq, startTime, startTime.Add(2*time.Millisecond)),
+				request(streamID, seq, startTime.Add(3*time.Millisecond), startTime.Add(4*time.Millisecond)),
+			},
+			ExpectedWitnesses: 1,
+			ExpectedLatencies: []float32{-4.0},
+		},
+		{
+			Name: "request + response zero timestamps",
+			PNTs: []akinet.ParsedNetworkTraffic{
+				request(streamID, seq, time.Time{}, time.Time{}),
+				response(streamID, seq, time.Time{}, time.Time{}),
+			},
+			ExpectedWitnesses: 1,
+			ExpectedLatencies: []float32{0.0},
+		},
+		{
+			Name: "response + request zero timestamps",
+			PNTs: []akinet.ParsedNetworkTraffic{
+				response(streamID, seq, time.Time{}, time.Time{}),
+				request(streamID, seq, time.Time{}, time.Time{}),
+			},
+			ExpectedWitnesses: 1,
+			ExpectedLatencies: []float32{0.0},
+		},
+		{
+			Name: "interleaved request + response pairs",
+			PNTs: []akinet.ParsedNetworkTraffic{
+				request(streamID, seq, startTime, startTime.Add(2*time.Millisecond)),
+				request(streamID, seq+1, startTime.Add(1*time.Millisecond), startTime.Add(3*time.Millisecond)),
+				response(streamID, seq, startTime.Add(5*time.Millisecond), startTime.Add(7*time.Millisecond)),
+				response(streamID, seq+1, startTime.Add(6*time.Millisecond), startTime.Add(8*time.Millisecond)),
+			},
+			ExpectedWitnesses: 2,
+			ExpectedLatencies: []float32{3.0, 3.0},
+		},
+		{
+			Name: "pipelined request + response pairs",
+			PNTs: []akinet.ParsedNetworkTraffic{
+				request(streamID, seq, startTime, startTime.Add(2*time.Millisecond)),
+				request(streamID, seq, startTime.Add(1*time.Millisecond), startTime.Add(3*time.Millisecond)),
+				response(streamID, seq, startTime.Add(5*time.Millisecond), startTime.Add(7*time.Millisecond)),
+				response(streamID, seq, startTime.Add(6*time.Millisecond), startTime.Add(8*time.Millisecond)),
+			},
+			ExpectedWitnesses: 2,
+			ExpectedLatencies: []float32{0.0, 0.0},
```

Comment:
This seems like a clear bug, maybe add a comment that this is recording the existing values but breaking it would be OK?